### DISCUSSION TITLE
Common functionalities wrapping

### DISCUSSION THREAD

**Discussion by FlorianPydde** [microsoft/dstoolkit-mlops-base#41] [Ideas]
As you see our github repositories, there are some similar functionalities like 
 
https://github.com/microsoft/dstoolkit-classification-solution-accelerator/blob/main/src/utils.py
and
https://github.com/microsoft/dstoolkit-mlops-base/blob/main/src/utils.py
 
They define common functions like generating `Workspace`, getting `Datasets` etc.

The proposition is to come up with a pypi package that utilizes common functionalities (see attached ppt)
[common_function_dstoolkit.pptx](https://github.com/microsoft/dstoolkit-mlops-base/files/7474364/common_function_dstoolkit.pptx)

**Comment by kyoro1** [comment]
@FlorianPydde , thanks for raising my opinion as discussion here. Yes, in order to avoid double maintenance, we need to sort out the common functions as standardized one.
As far as I assumed, `PyPI` should be the 1st candidate for it, I guessed.

**Reply by FlorianPydde** [reply]
Although it's a good idea, I see the following challenges: the util file is basically a wrapper around AML sdk functionalities. Hence, whoever maintains the creates maintains the package would have to monitor changes to the sdk.

**Reply by kyoro1** [reply]
That's right. The situation is the same, even if we adopt the current implementation, isn't it? In current situation, we have to monitor 2 repositories...
Of course, if we adopt them, please forget my suggestion. Every opinion has each merit/demerit in nature.

**Reply by michelole** [reply]
Can't we propose adding this upstream, i.e. into the AML SDK?
